api/src/main/java/org/openmrs/module/legacyui/api/impl/LegacyUIImpl.java

```diff
@@ -0,0 +1,291 @@
+import org.openmrs.api.APIException;
+import org.openmrs.api.OrderService;
+import org.openmrs.api.ProviderService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.impl.BaseOpenmrsService;
+import org.openmrs.module.legacyui.api.LegacyUIService;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Default implementation of the patient service. This class should not be used on its own. The
+ * current OpenMRS implementation should be fetched from the Context via
+ * <code>Context.getPatientService()</code>
+ * 
+ * @see org.openmrs.api.context.Context
+ * @see org.openmrs.api.PatientService
+ * @see org.openmrs.api.PersonService
+ */
+@Transactional
+public class LegacyUIImpl extends BaseOpenmrsService implements LegacyUIService {
+	
+	private final Log log = LogFactory.getLog(this.getClass());
+	
+	/**
+	 * Copied from OpenMRS core 1.12.x and added back #discontinueAllDrugOrders from 1.9.x
+	 * See https://github.com/openmrs/openmrs-core/blob/1.12.x/api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java#L1191
+	 * See https://github.com/openmrs/openmrs-core/blob/1.9.x/api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java#L1092
+	 * 
+	 * This is the way to establish that a patient has left the care center. This API call is
+	 * responsible for:
+	 * <ol>
+	 * <li>Closing workflow statuses</li>
+	 * <li>Terminating programs</li>
+	 * <li>Discontinuing orders</li>
+	 * <li>Flagging patient table</li>
+	 * <li>Creating any relevant observations about the patient (if applicable)</li>
+	 * </ol>
+	 * 
+	 * @param patient - the patient who has exited care
+	 * @param dateExited - the declared date/time of the patient's exit
+	 * @param reasonForExit - the concept that corresponds with why the patient has been declared as
+	 *            exited
+	 * @throws APIException
+	 */
+	public void exitFromCare(Patient patient, Date dateExited, Concept reasonForExit) throws APIException {
+		
+		if (patient == null) {
+			throw new APIException("Attempting to exit from care an invalid patient. Cannot proceed");
+		}
+		if (dateExited == null) {
+			throw new APIException("Must supply a valid dateExited when indicating that a patient has left care");
+		}
+		if (reasonForExit == null) {
+			throw new APIException(
+			        "Must supply a valid reasonForExit (even if 'Unknown') when indicating that a patient has left care");
+		}
+		
+		// need to create an observation to represent this (otherwise how
+		// will we know?)
+		saveReasonForExitObs(patient, dateExited, reasonForExit);
+		
+		// need to terminate any applicable programs
+		triggerStateConversion(patient, reasonForExit, dateExited);
+		
+		// need to discontinue any open orders for this patient
+		discontinueAllDrugOrders(patient, reasonForExit, dateExited);
+	}
+	
+	/**
+	 * Copied from OpenMRS core 1.12.x 
+	 * See https://github.com/openmrs/openmrs-core/blob/1.12.x/api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java#L1219
```

Comment:
Same as above for using tags

Reply:
[Fixed](https://github.com/openmrs/openmrs-module-legacyui/pull/128/commits/9dd92c8a5560377daedb2e231c639e315fa177b8)